yt_dlp/extractor/rheinmaintv.py

```diff
@@ -0,0 +1,601 @@
+                    'channels': 2,
+                    'bits_per_sample': 16,
+                    'nal_unit_length_field': 4
+                }
+            }],
+            'subtitles': {},
+            'thumbnail': 'https://rmtvmedia0.blob.core.windows.net/e1cf7655-91b9-4b7e-b19f-a79852c7b41f/06_Standard.jpg?sv=2016-05-31&sr=c&sig=jQPJSdxCxJMX8yUZkFUPNrv1qISb0pCouK9MzAMSOMc%3D&st=2022-11-14T14%3A50%3A02Z&se=3022-11-15T14%3A50%3A02Z&sp=r',
+            'timestamp': 1668527402,
+            'duration': 348.0,
+            'view_count': int,
+            'upload_date': '20221115'
+        }
+    }]
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        display_id = mobj.group('display_id')
+        video_id = 'vom %s' % mobj.group('date')
+        if mobj.group('serial_number'):
+            video_id += ' (%s)' % mobj.group('serial_number')
+        webpage = self._download_webpage(url, video_id)
+
+        headline = self._html_search_regex(r'<h1><span class="title">([^<]*)</span>', webpage, 'headline')
+
+        source, img = self._search_regex(r'(?s)(?P<source><source[^>]*>)(?P<img><img[^>]*>)', webpage, 'video', group=('source', 'img'))
+        source = extract_attributes(source)
+        img = extract_attributes(img)
+
+        # json = self._search_json_ld(webpage, video_id)  # uncomment to extract the useless 'contentUrl' (as 'url') instead of the essential 'embedUrl'
+        #                                                 # then use get(json, 'title') instead of get(json, 'name')
+        json = self._parse_json(self._search_regex(r'(?s)<script type="application/ld\+json">([^<]*)</script>', webpage, 'json'), video_id)
+        # json = None  # uncomment to go without ld+json altogether
+
+        def get(dictionary, *keys):  # return ..utils.traverse_obj(dictionary, keys)
+            for k in keys:
+                if dictionary is None:
+                    return None
+                dictionary = dictionary.get(k)
+            return dictionary
+
+        ism_manifest_url = source['src'] or get(json, 'embedUrl')  # attribute 'src' is mandatory
+        formats, subtitles = self._extract_ism_formats_and_subtitles(ism_manifest_url, video_id)
+
+        def removeprefix(string, prefix):
+            return string[len(prefix):] if string.startswith(prefix) else None
+
+        def extract_format(internet_media_type, media):  # subtype of media (aka type)
+            return removeprefix(internet_media_type, media + '/') if internet_media_type is not None else None
+
+        extension = extract_format(source.get('type'), 'video')  # use video format as filename extension (!?)
+        if extension:
+            for f in formats:
+                f['ext'] = extension
+
+        return {
+            'id': video_id,
+            'title': headline or img.get('title') or get(json, 'name') or self._og_search_title(webpage) or self._html_extract_title(webpage).removesuffix(' -'),
```

Comment:
```suggestion
            'title': (self._html_search_regex(r'<h1><span class="title">([^<]*)</span>',
                                              webpage, 'headline', default=None)
                      or img.get('title') or json.get('name')
                      or self._og_search_title(webpage, default=None)
                      or remove_end(self._html_extract_title(webpage), ' -')),
```

Reply:
I hate superfluous parentheses, especially if their purpose is to please a policy tool.

Reply:
The purpose of this policy is not to please a policy tool, but rather to create more readable code. Long lines are unreadable and the convention for this is to break it into more lines.

yt-dlp already is very lenient with its coding style so the conventions that it imposes should not be hard to follow.

Reply:
The lines won't get any longer without the parentheses.

Reply:
If you want to indent without additional parenthesis you can just do so:
```py
            'title':
                self._html_search_regex(r'<h1><span class="title">([^<]*)</span>',
                                        webpage, 'headline', default=None)
                or img.get('title') or json_ld.get('title')
                or self._og_search_title(webpage)
                or remove_end(self._html_extract_title(webpage), ' -'),
```
`str.removesuffix` is added with Python 3.9, so the `utils` function is used instead.

Reply:
Ah, finally this is a layout which flake8 accepts. Thank you!

I hope that the page title will never have to be extracted that way, but fixed that too.